src/ensembl/src/content/app/custom-download/containers/header/CustomDownloadHeader.tsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { PrimaryButton } from 'src/shared/button/Button';
+
+import RoundButton from 'src/shared/round-button/RoundButton';
+
+import { RootState } from 'src/store';
+import Select, { Option } from 'src/shared/select/Select';
+
+import {
+  getSelectedPreFilter,
+  getPreviewResult,
+  getShowPreviewResult,
+  getDownloadType
+} from '../../state/customDownloadSelectors';
+
+import { getFilters } from '../content/filter-accordion/state/filterAccordionSelector';
+import { getAttributes } from '../content/attributes-accordion/state/attributesAccordionSelector';
+
+import {
+  togglePreFiltersPanel,
+  setShowPreview,
+  setDownloadType
+} from '../../state/customDownloadActions';
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as BackIcon } from 'static/img/shared/chevron-left.svg';
+
+import {
+  getSelectedAttributes,
+  getSelectedFilters
+} from '../content/result-holder/helpers';
+
+import { fetchCustomDownloadResults } from './helper';
+
+import AttributesSection, {
+  SelectedAttribute
+} from 'src/content/app/custom-download/types/Attributes';
+
+import styles from './CustomDownloadHeader.scss';
+
+type Props = StateProps & DispatchProps;
+
+const downloadTypeoptions = [
+  {
+    label: 'CSV',
+    value: 'text/csv',
+    isSelected: false
+  },
+  {
+    label: 'JSON',
+    value: 'application/json',
+    isSelected: false
+  }
+  // {
+  //   label: 'CSV.gz',
+  //   value: 'text/csv',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'HTML',
+  //   value: 'application/json',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'TSV',
+  //   value: 'application/json',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'TSV.gz',
+  //   value: 'application/json',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'URL',
+  //   value: 'application/json',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'XML',
+  //   value: 'application/xml',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'XLS',
+  //   value: 'application/xml',
+  //   isSelected: false
+  // },
+  // {
+  //   label: 'XLS.gz',
+  //   value: 'application/xml',
+  //   isSelected: false
+  // }
+];
+
+const getFormattedTotal = (total: number) => {
+  if (!total) {
+    return 0;
+  }
+  return total.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+};
```

Comment:
What is this function doing? Is it formatting numbers in a way that adds commas as a thousands separator? If so, then it looks like a very generic function that deserves to be in one of the shared helpers (and deserves to be tested).

Reply:
Moved to shared folder and added some tests.